airbyte-cdk/python/airbyte_cdk/models/airbyte_protocol.py

```diff
@@ -293,6 +293,14 @@ class Config:
     global_: Optional[AirbyteGlobalState] = Field(None, alias="global")
     data: Optional[Dict[str, Any]] = Field(None, description="(Deprecated) the state data")
 
+    # To maintain backwards compatibility, all of AirbyteStateMessage's fields are optional, but we should validate
+    # that at least one of the three fields is defined.
+    @root_validator()
+    def check_at_least_one(cls, values):
+        if values.get("global_") is None and values.get("data") is None and values.get("stream") is None:
+            raise ValueError("AirbyteStateMessage should contain either a stream, global, or state field")
+        return values
```

Comment:
I think the protocol classes are auto-generated from the protocol definitions in YAML. This change will be removed on the next protocol update. A workaround could be to declare a `ValidatedStateType` inheriting from `Type` with this validator method.

Reply:
Really good callout! It might be a little heavy handed to define a new class and pass it everywhere for a pretty simple check. Instead I'm just going to do a conditional check in `read_state()` that returns a value error if input is invalid. And leave airbyte_protocol as is.